renderdoc/driver/vulkan/vk_pixelhistory.cpp

```diff
@@ -118,39 +126,216 @@ struct VulkanPixelHistoryCallback : public VulkanDrawcallCallback
       m_Events.insert(std::make_pair(events[i].eventId, events[i]));
   }
   ~VulkanPixelHistoryCallback() { m_pDriver->SetDrawcallCB(NULL); }
-  void CopyPixel(uint32_t eid, VkCommandBuffer cmd, size_t offset)
+  // CreateResources creates fixed colour shader, offscreen framebuffer and
+  // a renderpass for occlusion and stencil tests.
+  void CreateResources()
+  {
+    // Create fixed colour shader
+    float col[4] = {1.0, 0.0, 0.0, 1.0};
+    m_pDriver->GetDebugManager()->PatchFixedColShader(m_FixedColFS, col);
+
+    VkResult vkr;
+
+    // TODO: add depth/stencil attachment
+    VkAttachmentReference colRef = {0, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL};
+    VkSubpassDescription sub = {};
+    sub.pipelineBindPoint = VK_PIPELINE_BIND_POINT_GRAPHICS;
+    sub.colorAttachmentCount = 1;
+    sub.pColorAttachments = &colRef;
+
+    std::vector<VkAttachmentDescription> attachmentDescriptions;
+    VkAttachmentDescription colDesc = {};
+    colDesc.format = m_Format;
+    colDesc.samples = VK_SAMPLE_COUNT_1_BIT;
+    colDesc.loadOp = VK_ATTACHMENT_LOAD_OP_CLEAR;
+    colDesc.storeOp = VK_ATTACHMENT_STORE_OP_STORE;
+    colDesc.stencilLoadOp = VK_ATTACHMENT_LOAD_OP_DONT_CARE;
+    colDesc.stencilStoreOp = VK_ATTACHMENT_STORE_OP_DONT_CARE;
+    colDesc.initialLayout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+    colDesc.finalLayout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+    attachmentDescriptions.push_back(colDesc);
```

Comment:
If we know we only want one attachment, it would be preferable to just declare a single variable like `sub` above. Rather than using a `std::vector` which will heap allocate.

Reply:
Sorry, had this leftover from when I was adding a depth attachment, but at least not necessary for now. Removed.